static/app/components/contentSliderDiff/index.tsx

```diff
@@ -0,0 +1,264 @@
+import {Fragment, useRef} from 'react';
+import styled from '@emotion/styled';
+
+import NegativeSpaceContainer from 'sentry/components/container/negativeSpaceContainer';
+import QuestionTooltip from 'sentry/components/questionTooltip';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import toPixels from 'sentry/utils/number/toPixels';
+import {useDimensions} from 'sentry/utils/useDimensions';
+import {useResizableDrawer} from 'sentry/utils/useResizableDrawer';
+
+interface BeforeAfterProps {
+  label: React.ReactNode;
+  children?: React.ReactNode;
+  /**
+   * Provides additional info about what the diff is showing
+   */
+  help?: React.ReactNode;
+}
+
+function BeforeAfter({children, label, help}: BeforeAfterProps) {
+  return (
+    <Fragment>
+      <Label>
+        {label}
+        {help && <QuestionTooltip title={help} size="xs" />}
+      </Label>
+      {children}
+    </Fragment>
+  );
+}
+
+interface ContentSliderDiffProps {
+  /**
+   * The content to display after the divider. Usually an image or replay.
+   */
+  afterContent: React.ReactNode;
+  /**
+   * The content to display before the divider. Usually an image or replay.
+   */
+  beforeContent: React.ReactNode;
+  /**
+   * Provides additional info in the label's header about what the after diff is showing
+   */
+  afterHelp?: React.ReactNode;
+  /**
+   * Provides additional info in the label's header about what the before diff is showing
+   */
+  beforeHelp?: React.ReactNode;
+  minHeight?: `${number}px` | `${number}%`;
+  /**
+   * A callback function triggered when the divider is clicked (mouse down event).
+   * Useful when we want to track analytics.
+   */
+  onDividerMouseDown?: (e: React.MouseEvent) => void;
+}
+
+/**
+ * Compares the before and after of visual elements using an adjustable slider.
+ * It allows users to dynamically see the "before" and "after" sections by dragging a divider.
+ * The before and after contents are not directly defined here and have to be provided, so it can be very flexible
+ * (e.g. images, replays, etc).
+ */
+export function ContentSliderDiff({
+  beforeHelp,
+  afterHelp,
+  onDividerMouseDown,
+  beforeContent,
+  afterContent,
+  minHeight = '0px',
+}: ContentSliderDiffProps) {
+  const positionedRef = useRef<HTMLDivElement>(null);
+  const viewDimensions = useDimensions({elementRef: positionedRef});
+  const width = toPixels(viewDimensions.width);
+
+  return (
+    <Fragment>
+      <DiffHeader>
+        <BeforeAfter label={t('Before')} help={beforeHelp} />
+        <BeforeAfter label={t('After')} help={afterHelp} />
+      </DiffHeader>
```

Comment:
I'd probably move this out of the main `contentSliderDiff` and let people add it themselves. Including it hard-codes some assumptions about the labels, whether this goes above/below or gets hidden at all. The two props that it takes are totally independant from the rest of the slider which will make it easy to be it's own thing.